src/main/scala/nl.knaw.dans.easy.validatebag/rules/metadata/package.scala

```diff
@@ -237,9 +240,39 @@ package object metadata extends DebugEnhancedLogging {
     else Try(fail(s"Point with only one coordinate: ${ point.text.trim }"))
   }
 
+  def archisIdentifiersHaveAtMost10Characters(t: TargetBag): Try[Unit] = {
+    trace(())
+    for {
+      ddm <- t.tryDdm
+      identifiers = getArchisIdentifiers(ddm)
+      validationResult = identifiers.map(validateArchisIdentifier)
+      _ = fail(formatInvalidArchisIdentifiers(validationResult).mkString("\n"))
+    } yield ()
+  }
+
+  private def getArchisIdentifiers(ddm: Node): NodeSeq = {
+    (ddm \\ "identifier")
+      .filter(_.attributes
+        .filter(_.prefixedKey == "xsi:type")
+        .filter(_.value.text == "id-type:ARCHIS-ZAAK-IDENTIFICATIE")
+        .nonEmpty)
+  }
+
+  private def validateArchisIdentifier(identifier: Node): Try[Unit] = {
+    if (identifier.text.length <= 10) Success(())
+    else Try(fail(s"Archis identifier is too long: ${ identifier.text }"))
```

Comment:
Refer to the rule, e.g. "Archis identifier must not be longer than 10 characters" or something.

Reply:
Rephrased it to: `s"Archis identifier must be 10 or less characters long: $identifier"`. See c03111b.